dev/NavigationView/NavigationView.cpp

```diff
@@ -3632,3 +3820,159 @@ void NavigationView::UpdatePaneShadow()
         shadowReceiver.Margin(shadowReceiverMargin);
     }
 }
+
+template<typename T> T NavigationView::GetContainerForData(const winrt::IInspectable& data)
+{
+    if (!data)
+    {
+        return nullptr;
+    }
+
+    if (auto nvi = data.try_as<T>())
+    {
+        return nvi;
+    }
+
+    if (IsTopNavigationView())
```

Comment:
lots of duplication between the if and else blocks and some differences which probably should not be there, you could just pull this one line out and merge the blocks
auto repeater = IsTopNavigationView() ? m_topNavRepeater.get(): m_leftNavRepeater.get();